src/screens/Principal/components/BoardGame/index.js

```diff
@@ -0,0 +1,171 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import boardCells from '../constants/boardCells';
+import Block from '../Blocks/components/Block';
+
+import styles from './index.module.scss';
+
+class BoardGame extends React.Component {
+  state = {
+    dataBlocks: {},
+    dataBoardPositions: {},
+    lastColumnTemp: '',
+    updateCount: false
+  };
+
+  countFullColumns = () => {
+    const numberColumns = 16;
+    const dataBoardPositions = { ...this.state.dataBoardPositions };
+    const positions = Object.keys(dataBoardPositions).filter(columnId => dataBoardPositions[columnId] !== '');
+
+    if (positions.length === numberColumns) {
+      this.props.onDisabledButton({ disabled: false });
+    } else {
+      this.props.onDisabledButton({ disabled: true });
+    }
+  };
+
+  getBlock = (idBlock, idColumn) => {
+    if (idColumn !== '') {
+      const { state } = this;
+      const dataBlocks = { ...state.dataBlocks };
+      const dataBoardPositions = { ...state.dataBoardPositions };
+      let { transactions } = this.props;
+
+      transactions = JSON.stringify(transactions);
+
+      [dataBlocks[idColumn]] = JSON.parse(transactions).filter(transaction => transaction.uuid === idBlock);
+      dataBoardPositions[idColumn] = idBlock;
+
+      this.setState({
+        ...this.prevState,
+        dataBoardPositions,
+        dataBlocks
+      });
+    }
+  };
+
+  clearCell = idColumn => {
+    if (idColumn !== '') {
+      const { state } = this;
+      const dataBlocks = { ...state.dataBlocks };
+      const dataBoardPositions = { ...state.dataBoardPositions };
```

Comment:
Podés destructurar state en lugar de this
```suggestion
      const { dataBlocks, dataBoardPositions } = this.state;
      const dataBlocks = { ...dataBlocks };
      const dataBoardPositions = { ...dataBoardPositions };
```